File: data/grammar_pdfs/parse.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extractText(file) {
    let dataBuffer = fs.readFileSync(file);
    try {
        const data = await pdf(dataBuffer);
        const outFile = file.replace('.pdf', '_out.txt');
        fs.writeFileSync(outFile, data.text);
        console.log(`Successfully extracted ${data.text.length} characters to ${outFile}`);
    } catch (error) {
        console.error("Error reading PDF:", error);
    }
}

// Extract all pdfs in the current directory
const files = fs.readdirSync('.');
for (const file of files) {
    if (file.endsWith('.pdf')) {
        console.log(`Processing ${file}...`);
        extractText(file);
    }
}

File: data/grammar_pdfs/parse.py
import PyPDF2
import os

for file in os.listdir('.'):
    if file.endswith('.pdf'):
        print(f"Processing {file}...")
        try:
            with open(file, 'rb') as f:
                reader = PyPDF2.PdfReader(f)
                text = ""
                for page in reader.pages:
                    text += page.extract_text() + "\n"
                
                out_name = file.replace('.pdf', '_out.txt')
                with open(out_name, 'w', encoding='utf-8') as out_f:
                    out_f.write(text)
                print(f"Extraction successful: {len(text)} characters to {out_name}")
        except Exception as e:
            print(f"Error reading {file}: {e}")
